### PR TITLE
chore(build): port build process to Node.js

### PR DESCRIPTION
**Pull request by gkalpak**
Closes #198

<sub>I removed the old files in separate commit, in case we decide to keep them around (by ignoring that commit).</sub>

### CODE REVIEW

**Comment by petebacondarwin** [comment]
We will need to update Jenkins

**Comment by gkalpak** [comment]
How so ? Is there something that is not compatible with the current version - I could change that.

**Comment by petebacondarwin** [comment]
Simply because angularjs.org builds on Jenkins call `build.sh` :-P

**Comment by gkalpak** [comment]
Aha, I thought you meant we need to update the version of Jenkins itself :smile: 

BTW1: I have only tried this on Windows and Debian. Isn't CI running on OSX ? Can you test on a Mac (if not already done so :stuck_out_tongue:)

BTW2: The `src/.htaccess` file doesn't seem to be used, i.e. it isn't copied to `build/` and thus doesn't get commited to the `dist` branch. It doesn't seem to serve any other purpose either (but there might be other uses of the repo that I'm missing). So, I'm wondering if we actually need it :smile:

**Comment by petebacondarwin** [comment]
BTW2: you are probably correct here

**Comment by petebacondarwin** [comment]
I'll give the build a run on OS/X tonight

**Comment by petebacondarwin** [comment]
Is it possible to make all the async calls sync. If we can it would make the code much simpler.
Perhaps there is a HTTP call in there for CDN that prevents this?

**Comment by mgol** [comment]
I think it's a good practice to always use npm scripts in all CI commands. That way you may not change the server setup when changing task runners, various tools etc.

**Comment by petebacondarwin** [comment]
@mgol - is that comment referring to another comment in this thread?

**Comment by gkalpak** [comment]
I went for async (vs sync) calls, because (a) it is a good practice ( :stuck_out_tongue: ) and (b) most importantly because on Windows when trying to sync-delete directories I often get errors that the directory is not empty and cannot be deleted (even if it is empty). I assumed it is a timing issue/bug, but with async methods it worked every time (possibly by coincidence :stuck_out_tongue:).

File: scripts/build.js [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+function testBuild() {
+  announce(`Testing the current build (ENV: ${JSON.stringify(PTOR_ENV, null, 2)})...`);
+
+  const npmInstallCmd = `${utils.getExecutable('npm', true)} install`;
+  const wdrManagerCmd = `${utils.getExecutable('webdriver-manager')} update`;
+  const httpServerCmd = `${utils.getExecutable('http-server')} -p ${PTOR_PORT} ${DST_DIR}`;
+  const protractorCmd = `${utils.getExecutable('protractor')} ${PTOR_CONF}`;
+
+  const protractorOptions = {env: PTOR_ENV, stdio: 'inherit'};
```
Comment (petebacondarwin) [inline scripts/build.js]: Looks like this needs to be 

```js
const protractorOptions = {env: Object.assign({}, process.env, PTOR_ENV), stdio: 'inherit'};
```

for it to work on OS/X

**Comment by gkalpak** [comment]
@petebacondarwin , PTAL.

**Comment by petebacondarwin** [comment]
LGTM - squashing and merging

**Pull request closed without merging**